test/e2e/configurer/chain/queries_btclightclient.go

```diff
@@ -1,21 +1,56 @@
 package chain
 
 import (
+	"encoding/base64"
 	"net/url"
+	"strconv"
 
-	"github.com/stretchr/testify/require"
+	"github.com/cosmos/cosmos-sdk/types/query"
+	sdkquerytypes "github.com/cosmos/cosmos-sdk/types/query"
 
 	"github.com/babylonlabs-io/babylon/test/e2e/util"
 	btclighttypes "github.com/babylonlabs-io/babylon/x/btclightclient/types"
 )
 
-func (n *NodeConfig) QueryBtcLightClientMainchain() []*btclighttypes.BTCHeaderInfoResponse {
-	bz, err := n.QueryGRPCGateway("/babylon/btclightclient/v1/mainchain", url.Values{})
-	require.NoError(n.t, err)
+func (n *NodeConfig) QueryBtcLightClientMainchain(pagination *query.PageRequest) (*btclighttypes.QueryMainChainResponse, error) {
+	queryParams := url.Values{}
+	if pagination != nil {
+		queryParams.Set("pagination.key", base64.URLEncoding.EncodeToString(pagination.Key))
+		queryParams.Set("pagination.limit", strconv.Itoa(int(pagination.Limit)))
+	}
+
+	bz, err := n.QueryGRPCGateway("/babylon/btclightclient/v1/mainchain", queryParams)
+	if err != nil {
+		return nil, err
+	}
 
 	var resp btclighttypes.QueryMainChainResponse
 	err = util.Cdc.UnmarshalJSON(bz, &resp)
-	require.NoError(n.t, err)
+	if err != nil {
+		return nil, err
+	}
+
+	return &resp, nil
+}
+
+func (n *NodeConfig) QueryBtcLightClientMainchainAll() []*btclighttypes.BTCHeaderInfoResponse {
+	headers := make([]*btclighttypes.BTCHeaderInfoResponse, 0)
+
+	limit := uint64(1000)
+	pagination := &sdkquerytypes.PageRequest{
+		Limit: limit,
+	}
+	for {
+		resp, err := n.QueryBtcLightClientMainchain(pagination)
+		if err != nil {
+			// err could come as {"code":3,"message":"header specified by key does not exist","details":[]}
```

Comment:
seems errors will be hidden if the error is not the expected one

Reply:
modified to check err